cmd/influxd/run/server.go

```diff
@@ -131,19 +131,26 @@ func NewServer(c *Config, buildInfo *BuildInfo) (*Server, error) {
 		err:       make(chan error),
 		closing:   make(chan struct{}),
 
-		BindAddress: bind,
-
-		MetaClient: meta.NewClient(c.Meta),
-
-		Monitor: monitor.New(c.Monitor),
-
+		BindAddress:       bind,
+		MetaClient:        meta.NewClient(c.Meta),
+		Monitor:           monitor.New(c.Monitor),
 		reportingDisabled: c.ReportingDisabled,
+		tcpAddr:           bind,
+		config:            c,
+	}
 
-		httpAPIAddr: c.HTTPD.BindAddress,
-		httpUseTLS:  c.HTTPD.HTTPSEnabled,
-		tcpAddr:     bind,
+	for _, i := range c.HTTPDInputs {
+		if i.BindAddress != "" {
```

Comment:
So first listed HTTP input is the one we use for any code that returns an HTTP address?